.github/workflows/dotnetcore.yml

```
@@ -19,11 +19,16 @@ jobs:
     - name: Setup .NET Core
       uses: actions/setup-dotnet@v1
       with:
-        dotnet-version: 6.0.x
+        dotnet-version: 7.0.x
     - name: Build with dotnet
       run: dotnet build --configuration Release
     - name: Test with dotnet test
       run: dotnet test --configuration Release --no-build
+    - name: Nerdbank.GitVersioning
+      uses: dotnet/nbgv@master
+      with:
+        setAllVars: true
+    - run: echo "Package generated artefacts/DotNetDockerTest.${NBGV_NuGetPackageVersion}.nupkg"
     - name: Publish to NuGet
       if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
-      run: dotnet nuget push artefacts/DotNetDockerTest.${GITHUB_REF#refs/tags/v}.nupkg --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
+      run: dotnet nuget push artefacts/DotNetDockerTest.${{ env.NBGV_NuGetPackageVersion }}.nupkg --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
```

I believe this could be simplified like this:
```suggestion
      run: dotnet nuget push artefacts/DotNetDockerTest.${NBGV_NuGetPackageVersion }.nupkg --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
```

---

I tried this on my branch and failed. 
https://github.com/wo-roberts/DotNetDockerTest/actions/runs/3970433336/jobs/6806147689
I will confess bash replacement rules are eluding me so I've left as is for now :) .

---

I don't think it failed for that reason. As a test, you could try prefixing the command with `echo` just to confirm that the variable replacement does occur.

It's a total nitpick anyway, and your current version does get the job done.